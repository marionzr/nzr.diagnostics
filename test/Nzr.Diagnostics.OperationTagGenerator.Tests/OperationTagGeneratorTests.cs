using FluentAssertions;
using Microsoft.EntityFrameworkCore;
using Nzr.Diagnostics.Testing.TestDatabaseSupport;
using Snapshooter.Xunit;

namespace Nzr.Diagnostics.OperationTagGenerator.Tests;

[Collection(TestSQLiteDbContextCollection.CollectionName)]
public class OperationTagGeneratorTests : IAsyncLifetime
{

    private readonly TestSQLiteDbContextFixture _fixture;

    public OperationTagGeneratorTests(TestSQLiteDbContextFixture fixture)
    {
        _fixture = fixture;
    }

    public async Task DisposeAsync()
    {
        await _fixture.ResetDatabaseAsync();
    }

    public async Task InitializeAsync()
    {
        await _fixture.ResetDatabaseAsync();
    }

    [Fact]
    public void NewTag_ShouldGenerateValidOperationTag()
    {
        // Act

        var operationTag = TagGenerator.NewTag();

        // Assert

        operationTag.Should().MatchSnapshot();
    }

    [Fact]
    public void NewTag_Should_Be_Included_In_Linq_GeneratedSql()
    {
        // Arrange

        using var context = _fixture.GetDbContext();

        // Act

        var query = context.Products
            .Where(p => p.Price > 10)
            .TagWith(TagGenerator.NewTag());

        var sql = query.ToQueryString();

        // Assert

        sql.Should().MatchSnapshot();
    }

    [Fact]
    public void NewTag_Should_Be_Included_In_Raw_GeneratedSql()
    {
        // Arrange

        using var context = _fixture.GetDbContext();

        // Act

        var query = context.Products
            .FromSqlRaw("SELECT * FROM Products WHERE Price > 10")
            .TagWith(TagGenerator.NewTag());

        var sql = query.ToQueryString();

        // Assert

        sql.Should().MatchSnapshot();
    }
}
