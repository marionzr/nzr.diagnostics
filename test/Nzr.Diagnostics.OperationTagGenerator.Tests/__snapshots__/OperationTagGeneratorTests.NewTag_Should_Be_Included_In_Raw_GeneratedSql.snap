-- Assembly: Nzr.Diagnostics.OperationTagGenerator.Tests
-- File: OperationTagGeneratorTests.cs
-- Member: NewTag_Should_Be_Included_In_Raw_GeneratedSql
-- Line: 72

SELECT * FROM Products WHERE Price > 10
